server/src/main/java/org/elasticsearch/action/support/replication/TransportWriteAction.java

```diff
@@ -61,21 +62,25 @@
         > extends TransportReplicationAction<Request, ReplicaRequest, Response> {
 
     private final IndexingPressure indexingPressure;
-    private final String executor;
+    private final Function<IndexShard, String> executor;
```

Comment:
Nitpick: the name `executor` isn't really right now that it's actually a function to get an executor.